### PR TITLE
fix: SwiftLint warnings for multiple rules

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR I fix warnings and errors related to the SwiftLint rules below:

- **Computed Accessors Order Violation**: Computed properties should declare first the getter and then the setter. `(computed_accessors_order)`
- **Operator Function Whitespace Violation**: Operators should be surrounded by a single whitespace when defining them. `(operator_whitespace)`
- **Returning Whitespace Violation**: Return arrow and return type should be separated by a single space or on a separate line. `(return_arrow_whitespace)`
**-Implicit Getter Violation**: Computed read-only properties should avoid using the get keyword. `(implicit_getter)`
- Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. `(vertical_parameter_alignment)`
- **Redundant String Enum Value Violation**: String enum values can be omitted when they are equal to the enumcase name. `(redundant_string_enum_value)`
- **For Where Violation**: `where` clauses are preferred over a single `if` inside a `for`. `(for_where)`
- **Unused Optional Binding Violation**: Prefer `!= nil` over `let _ =` `(unused_optional_binding)`
- **Orphaned Doc Comment Violation**: A doc comment should be attached to a declaration. `(orphaned_doc_comment)`
- **Colon Spacing Violation**: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. `(colon)`
- **Trailing Semicolon Violation**: Lines should not have trailing semicolons. `(trailing_semicolon)`
- **Vertical Whitespace Violation**: Limit vertical whitespace to a single empty line. Currently 3. `(vertical_whitespace)`
- **Trailing Whitespace Violation**: Lines should not have trailing whitespace. `(trailing_whitespace)`
- **NSObject Prefer isEqual Violation**: NSObject subclasses should implement isEqual instead of ==.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by sb88k — APPROVED** [review]

File: Source/Model/Label/Label.swift [hunk 1]
```diff
@@ -109,11 +109,10 @@ public class Label: ZMManagedObject, LabelType {
             fatal("Label entity not registered in managed object model")
         }
 
-        for managedObject in context.registeredObjects {
-            if managedObject.entity == entity, !managedObject.isFault {
+        for managedObject in context.registeredObjects where managedObject.entity == entity && !managedObject.isFault {
                 guard let label = managedObject as? Label, label.kind == .favorite else { continue }
                 return label
-            }
+
```
Comment (johnxnguyen) [inline Source/Model/Label/Label.swift:115]: The indentation in the for loop seems off now.

File: Source/Model/Message/ConversationMessage.swift [hunk 2]
```diff
@@ -325,10 +325,8 @@ extension ZMMessage {
 
     @objc public var usersReaction: [String: [UserType]] {
         var result = [String: [ZMUser]]()
-        for reaction in reactions {
-            if reaction.users.count > 0 {
+        for reaction in reactions where reaction.users.count > 0 {
```
Comment (johnxnguyen) [inline Source/Model/Message/ConversationMessage.swift:328]: Also here the indentation in the loop needs fixing.

File: Source/Model/User/ZMUser+RichProfile.swift [hunk 3]
```diff
@@ -30,10 +30,6 @@ public enum ZMUserKeys {
         self.value = value
     }
 
-    public static func == (lhs: UserRichProfileField, rhs: UserRichProfileField) -> Bool {
-        return lhs.type == rhs.type && lhs.value == rhs.value
-    }
-
```
Comment (johnxnguyen) [inline Source/Model/User/ZMUser+RichProfile.swift:36]: What is the lint rule that caused this change?

File: Tests/Source/Model/Conversation/ZMConversationTests.swift [hunk 4]
```diff
@@ -87,7 +87,7 @@ extension ZMConversationTests {
         let result = uiMOC.executeFetchRequestOrAssert(request)
 
         XCTAssertEqual(result.count, 1)
-        /// TODO
+        // TODO
```
Comment (johnxnguyen) [inline Tests/Source/Model/Conversation/ZMConversationTests.swift:90]: You can just delete this, it doesn't tell us what is to be done.

File: Source/Model/User/ZMUser+RichProfile.swift (hunk 3 above)
Comment (agisilaos) [reply Source/Model/User/ZMUser+RichProfile.swift:36]: @johnxnguyen NSObject Prefer isEqual: https://realm.github.io/SwiftLint/nsobject_prefer_isequal.html

Comment (johnxnguyen) [reply Source/Model/User/ZMUser+RichProfile.swift:36]: ok, the question is whether by removing this function the behaviour of `isEqual` stays the same.

Comment (agisilaos) [reply Source/Model/User/ZMUser+RichProfile.swift:36]: I think it does?! cc: @sb88k I remember we had a discussion about this

```swift
public override func isEqual(_ object: Any?) -> Bool {
         guard let other = object as? UserRichProfileField else { return false }
         return self == other
```

Comment (sb88k) [reply Source/Model/User/ZMUser+RichProfile.swift:36]: As SwiftLint suggests, if == operator is replaced by `isEqual`, it works as expected.

```swift
public override func isEqual(_ object: Any?) -> Bool {
     guard let other = object as? UserRichProfileField else { return false }
     return self.type == other.type && self.value == other.value
}
```

**Review by sb88k — COMMENTED** [review]
Looks like there is some mis-replacements.

File: Source/Model/User/ZMUser+RichProfile.swift (hunk 3 above)
Comment (agisilaos) [reply Source/Model/User/ZMUser+RichProfile.swift:36]: Ahhh, I see! Good catch. Thank you, I will update it.

Comment (agisilaos) [reply Source/Model/User/ZMUser+RichProfile.swift:36]: Done!

**Review by johnxnguyen — APPROVED** [review]

**Review by sb88k — APPROVED** [review]

**Pull request merged**